index.js

```diff
@@ -21,42 +21,43 @@ export default async function pMap(
 		const errors = [];
 		const skippedIndexes = [];
 		let isRejected = false;
+		let isRejectedOrResolved = false;
 		let isIterableDone = false;
 		let resolvingCount = 0;
 		let currentIndex = 0;
-		let asyncIterator = false;
-		let iterator;
-
-		if (iterable[Symbol.iterator] === undefined) {
-			// We've got an async iterable
-			iterator = iterable[Symbol.asyncIterator]();
-			asyncIterator = true;
-		} else {
-			iterator = iterable[Symbol.iterator]();
-		}
+		const iterator = iterable[Symbol.iterator] === undefined ? iterable[Symbol.asyncIterator]() : iterable[Symbol.iterator]();
 
 		const reject = reason => {
 			isRejected = true;
+			isRejectedOrResolved = true;
```

Comment:
This can be named `isResolved` as resolving means it either rejected or fulfilled.

Reply:
Done